src/EFCore.PG/Query/ExpressionTranslators/Internal/NpgsqlDateAddTranslator.cs

```diff
@@ -86,32 +87,48 @@ public virtual Expression Translate(MethodCallExpression methodCallExpression)
             if (!MethodInfoDatePartMapping.TryGetValue(methodCallExpression.Method, out var datePart))
                 return null;
 
-            // Ideally we would use Expression.Add() to have a simple plus-sign expression generated for us.
-            // But to represent an interval we'd need to use a TimeSpan, which does not represent months/years.
-            // We could use the provider-specific NpgsqlTimeSpan but it's not currently supported and is somewhat
-            // ugly, so we just generate the expression ourselves with CustomBinaryExpression
+            // When we have a constant, we create the interval via INTERVAL '1 days'. But this expression cannot be
+            // parameterized, so we use CAST($1 || ' days' AS interval) instead
+            // Note that a make_interval() function also exists, but accepts only int (for all fields except for
+            // seconds), so we don't use it.
+
+            Expression interval = null;
 
-            Expression instance = methodCallExpression.Object;
-            Expression amountToAdd = methodCallExpression.Arguments[0];
+            var instance = methodCallExpression.Object;
+            var amountToAdd = methodCallExpression.Arguments[0];
 
             if (instance is null || amountToAdd is null)
                 return null;
 
-            if (amountToAdd is ConstantExpression amount &&
-                amount.Type == typeof(double) &&
-                ((double)amount.Value >= int.MaxValue ||
-                 (double)amount.Value <= int.MinValue))
-                return null;
+            if (amountToAdd is ConstantExpression constantExpression)
+            {
+                if (constantExpression.Type == typeof(double) &&
+                    ((double)constantExpression.Value >= int.MaxValue ||
+                     (double)constantExpression.Value <= int.MinValue))
+                {
```

Comment:
Could omit the braces

Reply:
I prefer to keep them in cases of multiline conditions like this, IMHO it improves readability